client/src/pages/Admin.js

```diff
@@ -1,43 +1,63 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from "react";
-import { useState,useEffect } from "react";
+import { useState, useEffect } from "react";
 
 function AdminPage() {
-  const [pages, setPages] = useState([]);
+  const [pagesData, setPagesData] = useState([]);
 
-  async function fetchPagesAPI() {
-    const page = await fetch("/api/pages");
-    const response = await page.json();
-    setPages(response);
-	
+  async function fectPageTitles() {
+    try {
+      const getPageTitles = await fetch("/api/pages");
+      const allPageTitles = await getPageTitles.json();
+      return allPageTitles;
+    } catch (error) {
+      console.error(error);
+      return "Page TITLES  are not coming!! investigate why";
+    }
   }
- 
+
+  async function fectPagesData(pageTitle) {
+    try {
+      const getPagesData = await fetch(`api/pages/${pageTitle}`);
+      const pagesData = await getPagesData.json();
+      return pagesData;
+    } catch (error) {
+      console.error(error);
+      return "Pages DATA are not  coming!! investigate why";
+    }
+  }
+
   useEffect(() => {
-     fetchPagesAPI();
+    fectPageTitles()
+      .then((pagesTitleResults) => {
+        const pageDetails = Promise.all(
+          pagesTitleResults.map((page) => fectPagesData(page.page_title))
+        );
+        return pageDetails;
+      })
+      .then((pagesData) => setPagesData(pagesData));
   }, []);
-  
-	return (
-		<div>
-            
-            {pages.map((item)=>{return<div key={item.page_id}> <h1>{item.page_title}</h1>
-               <ul>
-               <li>Header</li>
-               <li>Module 1</li>
-               <li>Module 2</li>
-               <li>Footer</li>
-             </ul>
-             <label>Add a module:</label>
-       
-       <select >
-         <option >Module 1</option>
-         <option >Module 2</option>
-       </select></div>})}
- 
-
-
-<button>Add page</button>
-		</div>
-	);
+
+  if (!pagesData) <p>Loading..</p>;
+
+  console.log(pagesData.map((eachPage) => console.log(eachPage.modules)));
```

Comment:
remove all console.logs (you can keep the console.errors) because they are only for debugging purposes.

Also, I noticed you removed 'add module' and 'add page'
I know they are static and they don't do anything. but please put them back for the future tickets

Reply:
Done